activerecord/lib/active_record/associations.rb

```diff
@@ -1667,6 +1667,13 @@ def has_many(name, scope = nil, **options, &extension)
         #
         #   Note that NestedAttributes::ClassMethods#accepts_nested_attributes_for sets
         #   <tt>:autosave</tt> to <tt>true</tt>.
+        
```

Comment:
I think it makes more sense to add this in the `validate` section above.

Also, while you are at it, the existing `+false+ by default.` statement is a bit misleading. Since the default depends on the `autosave` value, saying it's `false` by default makes little sense. The examples you gave in your patch are more clear.